server/src/main/java/tabletop/controllers/TournamentController.java

```diff
@@ -21,105 +26,104 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(TournamentController.class);
 
     private TournamentService tournamentService;
+    private UserService userService;
 
-    public TournamentController(TournamentService tournamentService) {
+    public TournamentController(TournamentService tournamentService, UserService userService) {
         this.tournamentService = tournamentService;
+        this.userService = userService;
     }
 
     @RequestMapping(value = "/types", method = RequestMethod.GET)
-    public List<TournamentType> getTournamentTypesList() {
-        List<TournamentType> tournamentTypes = new ArrayList<>();
-        tournamentTypes.addAll(Arrays.asList(TournamentType.values()));
-        return tournamentTypes;
+    public ResponseEntity<List<TournamentType>> getTournamentTypesList() {
+        return ResponseEntity.ok(Arrays.asList(TournamentType.values()));
     }
 
     @RequestMapping(value = "/finished", method = RequestMethod.GET)
-    public Collection<Tournament> getFinishedTournaments() {
-        return tournamentService.getFinishedTournaments();
+    public ResponseEntity<Collection<Tournament>> getFinishedTournaments() {
+        return ResponseEntity.ok(tournamentService.getFinishedTournaments());
     }
 
     @RequestMapping(value = "/show/{tournamentid}", method = RequestMethod.GET)
-    public Tournament getTournamentById(@PathVariable("tournamentid") Long tournamentid) {
+    public ResponseEntity<Tournament> getTournamentById(@PathVariable("tournamentid") Long tournamentid) {
```

Comment:
nice with the auto conversion here, I was converting a String to Long. If it doesn't match it uses the handleTypeMismatch in ControllerExceptionHandler I made a while ago.